### PR TITLE
feat: add separated grades bucket and manage querystring by config var

### PR DESCRIPTION
**Pull request by Henrrypg**
# Description

This PR pretends to add separated bucket for grades because the data store there can be sensitive.

Tutor-minio currently have the [default bucket public](https://github.com/overhangio/tutor-minio/blob/master/tutorminio/templates/minio/tasks/minio/init.sh#L6) and the other ones private, the idea of this PR is set `AWS_QUERYSTRING_AUTH` editable and `True` by default, because the querystring is needed to access to private buckets.

> If [custom_domain](https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto3.py#L604) is send, it will return the url before add the [querystring](https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto3.py#L615). But with this change we need to set `AWS_S3_SECURE_URLS` because s3 [enforces https method](https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto3.py#L255).

# How to test it

You can install this branch and with a basic tutor installation test the changes. Here is some screenshots of mi own test on local.

1. Grades bucket is private
image

2. I can download the file without problem, in the url we can look the querystring 
image

3. I can upload and look the profile image
image

4. For local purposes, the method is `http` and no querystring is sent as expected
image

### CODE REVIEW

File: tutorminio/patches/openedx-common-settings [hunk 1]
```diff
@@ -2,6 +2,7 @@ DEFAULT_FILE_STORAGE = "storages.backends.s3boto3.S3Boto3Storage"
 VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"].lstrip("/")
 VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"].lstrip("/")
 GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"] = GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"].lstrip("/")
+GRADES_DOWNLOAD["STORAGE_KWARGS"]["bucket"] = "{{ MINIO_GRADE_BUCKET }}"
```
Comment (regisb) [inline tutorminio/patches/openedx-common-settings:5]: Can you please explain what will be the impact of this change? Am I right to understand that files stored in the grades download bucket are generated when the instructor generates the course grades? If yes, then you're right that this bucket should be private by default.

File: tutorminio/patches/openedx-lms-common-settings [hunk 2]
```diff
@@ -5,5 +5,6 @@ PROFILE_IMAGE_BACKEND = {
         "location": PROFILE_IMAGE_BACKEND["options"]["location"].lstrip("/"),
         # the following non empty property is necessary in development
         "base_url": "dummyprofileimagebaseurl",
+        "custom_domain": "{{ MINIO_HOST }}/{{ MINIO_BUCKET_NAME }}"
```
Comment (regisb) [inline tutorminio/patches/openedx-lms-common-settings:8]: Why is this change necessary?

File: tutorminio/patches/openedx-common-settings (hunk 1 above)
Comment (Henrrypg) [reply tutorminio/patches/openedx-common-settings:5]: Hello @regisb, yes, the files stored there are these generated by instructutor like Grades Download, it can have sensitive data

File: tutorminio/patches/openedx-auth [hunk 3]
```diff
@@ -1,3 +1,3 @@
 AWS_STORAGE_BUCKET_NAME: "{{ MINIO_BUCKET_NAME }}"
 AWS_S3_CUSTOM_DOMAIN: ""
-AWS_QUERYSTRING_AUTH: false
\ No newline at end of file
+AWS_QUERYSTRING_AUTH: {{ MINIO_QUERYSTRING_AUTH }}
```
Comment (regisb) [inline tutorminio/patches/openedx-auth:3]: This will render to "False", and not "false". (note the upper case "F") "False" will be incorrectly evaluated to "true" in Python code. 

I think you need:

    {% if MINIO_QUERYSTRING_AUTH %}true{% else %}false {% endif %}

But then I'm wondering what's the purpose of the `MINIO_QUERYSTRING_AUTH`. What role is it playing for you?

Comment (regisb) [reply tutorminio/patches/openedx-auth:3]: My comment above is still valid. The `MINIO_QUERYSTRING_AUTH` variable will not render to "false" or "true".

**Review by regisb — CHANGES_REQUESTED** [review]
I understand that the `AWS_QUERYSTRING_AUTH` change could be treated in a separate PR, distinct from the topic of the separate grades bucket. But my understanding is that they are actually connected.

I see one very good reason to use a separate bucket for the grades: that the grades bucket should be private while the others should be public.

1. When we run standalone MinIO (i.e: not in gateway mode) how do we create a private bucket in MinIO?
2. How do we access this private bucket from the LMS?
3.

**Comment by Henrrypg** [comment]
Hello @regisb i'm sorry for the late, i just commit a couple of changes about this PR and your suggestions, and i also change the description if this PR explaining the changes and some screenshots of how is working.

**Comment by FahadKhalid210** [comment]
Hi @Henrrypg, I will look into it.
Can you please resolve the conflicts?

**Comment by FahadKhalid210** [comment]
@Henrrypg please resolve [this](https://github.com/overhangio/tutor-minio/pull/26#pullrequestreview-1637488402)

**Comment by FahadKhalid210** [comment]
@Henrrypg I'm following up on my previous message regarding the need for resolution on [this](https://github.com/overhangio/tutor-minio/pull/26#pullrequestreview-1637488402). It's been three weeks since I reached out, and I'm eager to address this matter. Your prompt attention to this would be greatly appreciated.

**Comment by Henrrypg** [comment]
Hello @FahadKhalid210 I'm sorry for the late.

Answering the questions:

> 1. When we run standalone MinIO (i.e: not in gateway mode) how do we create a private bucket in MinIO?

Buckets are create private by default, [here](https://github.com/overhangio/tutor-minio/blob/master/tutorminio/templates/minio/tasks/minio/init.sh#L6) we set the public policy only to `MINIO_BUCKET_NAME`

> 2. How do we access this private bucket from the LMS?

We can access to the private buckets via `MINIO_QUERYSTRING_AUTH`, if this variable is set to True, a querystring is sent in each call to a bucket and it authenticates the request

File: tutorminio/patches/openedx-common-settings [hunk 4]
```diff
@@ -2,6 +2,7 @@ DEFAULT_FILE_STORAGE = "storages.backends.s3boto3.S3Boto3Storage"
 VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_IMAGE_SETTINGS["STORAGE_KWARGS"]["location"].lstrip("/")
 VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"] = VIDEO_TRANSCRIPTS_SETTINGS["STORAGE_KWARGS"]["location"].lstrip("/")
 GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"] = GRADES_DOWNLOAD["STORAGE_KWARGS"]["location"].lstrip("/")
+GRADES_DOWNLOAD["STORAGE_KWARGS"]["bucket"] = "{{ MINIO_GRADES_BUCKET_NAME }}"
```
Comment (FahadKhalid210) [inline tutorminio/patches/openedx-common-settings:5]: The `bucket` option has been [deprecated and renamed](https://github.com/jschneier/django-storages/blob/70fa8d96665f0c5e44aacd6c782f6489fa8dc11e/CHANGELOG.rst#L443) to `bucket_name`.

File: tutorminio/plugin.py [hunk 5]
```diff
@@ -24,6 +24,8 @@
         "VIDEO_UPLOAD_BUCKET_NAME": "openedxvideos",
         "HOST": "files.{{ LMS_HOST }}",
         "CONSOLE_HOST": "minio.{{ LMS_HOST }}",
+        "GRADES_BUCKET_NAME": "openedxgrades",
+        "QUERYSTRING_AUTH": False,
```
Comment (FahadKhalid210) [inline tutorminio/plugin.py:28]: Since openedxgrades bucket is private so `MINIO_QUERYSTRING_AUTH` should be `true` by default otherwise we will get error when trying to download csv 
image

**Review by FahadKhalid210 — CHANGES_REQUESTED** [review]

File: tutorminio/patches/openedx-common-settings (hunk 4 above)
Comment (Henrrypg) [reply tutorminio/patches/openedx-common-settings:5]: I'll address it

File: tutorminio/patches/openedx-auth [hunk 6]
```diff
@@ -1,3 +1,4 @@
 AWS_STORAGE_BUCKET_NAME: "{{ MINIO_BUCKET_NAME }}"
 AWS_S3_CUSTOM_DOMAIN: ""
-AWS_QUERYSTRING_AUTH: false
\ No newline at end of file
+AWS_QUERYSTRING_AUTH: {% if MINIO_QUERYSTRING_AUTH %}true{% else %}false{% endif %}
+AWS_S3_SECURE_URLS: {{ ENABLE_HTTPS }}
```
Comment (regisb) [inline tutorminio/patches/openedx-auth:4]: This is going to be rendered as:

```
AWS_S3_SECURE_URLS: True
```

or:

```
AWS_S3_SECURE_URLS: False
```

But "True" and "False" (with uppercase) are not yaml booleans. They are strings. Which will mean that `AWS_S3_SECURE_URLS` will always evaluate to "true".

**Comment by FahadKhalid210** [comment]
@Henrrypg please add a changelog entry with `scriv create`.

**Comment by Henrrypg** [comment]
Donw @FahadKhalid210  :)

**Review by FahadKhalid210 — APPROVED** [review]
LGTM 👍
